pkg/controller/tailoredprofile/tailoredprofile_controller.go

```diff
@@ -254,39 +343,64 @@ func (r *ReconcileTailoredProfile) Reconcile(ctx context.Context, request reconc
 
 	}
 
-	rules, ruleErr := r.getRulesFromSelections(instance, pb)
-	if ruleErr != nil && !common.IsRetriable(ruleErr) {
-		// Surface the error.
-		suerr := r.handleTailoredProfileStatusError(instance, ruleErr)
-		return reconcile.Result{}, suerr
-	} else if ruleErr != nil {
-		return reconcile.Result{}, ruleErr
-	}
+	// we should do the following only if the tailored profile does not have CustomRules
+	if len(customRules) <= 0 {
+		rules, ruleErr := r.getRulesFromSelections(instance, pb)
+		if ruleErr != nil && !common.IsRetriable(ruleErr) {
+			// Surface the error.
+			suerr := r.handleTailoredProfileStatusError(instance, ruleErr)
+			return reconcile.Result{}, suerr
+		} else if ruleErr != nil {
+			return reconcile.Result{}, ruleErr
+		}
 
-	if ruleValidErr := assertValidRuleTypes(rules); ruleValidErr != nil {
-		// Surface the error.
-		suerr := r.handleTailoredProfileStatusError(instance, ruleValidErr)
-		return reconcile.Result{}, suerr
-	}
+		if ruleValidErr := assertValidRuleTypes(rules); ruleValidErr != nil {
+			// Surface the error.
+			suerr := r.handleTailoredProfileStatusError(instance, ruleValidErr)
+			return reconcile.Result{}, suerr
+		}
+
+		variables, varErr := r.getVariablesFromSelections(instance, pb)
+		if varErr != nil && !common.IsRetriable(varErr) {
+			// Surface the error.
+			suerr := r.handleTailoredProfileStatusError(instance, varErr)
+			return reconcile.Result{}, suerr
+		} else if varErr != nil {
+			return reconcile.Result{}, varErr
+		}
+		tpcm := newTailoredProfileCM(instance)
 
-	variables, varErr := r.getVariablesFromSelections(instance, pb)
-	if varErr != nil && !common.IsRetriable(varErr) {
-		// Surface the error.
-		suerr := r.handleTailoredProfileStatusError(instance, varErr)
-		return reconcile.Result{}, suerr
-	} else if varErr != nil {
-		return reconcile.Result{}, varErr
+		tpcm.Data[tailoringFile], err = xccdf.TailoredProfileToXML(instance, p, pb, rules, variables)
```

Comment:
All these SCAP specific scanning details would be encapsulated in a SCAP handler, but behind a generic method name (e.g., `handler.prepScan()`).